lib/adapter.js

```diff
@@ -0,0 +1,494 @@
+/*******************************************************************************
+ * Copyright 2014, 2015, 2016 IBM Corp.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
we will look into this and check with project management as to what license we should be using